bench/benchmark2html.Rmd

```diff
@@ -0,0 +1,93 @@
+---
+title: '{gtsummary} benchmark'
+author: "Gustavo Zapata Wainberg & Daniel Sjoberg"
+output: html_document
+---
+
+
+Updated: `r format(Sys.Date(), "%B %d %Y")`  
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+
+library(gert)
+library(dplyr)
+# library(stringr)
+library(ggplot2)
+library(bench) # requires dev version!
+library(plotly)
+
+bench::cb_fetch(remote = "upstream")
+
+# Get the data with previous benchmarks
+stored_benchmarks <- bench::cb_read()
+
+# commitInfo <- gert::git_commit_info() # Current commit (?).
+# commitInfo$id # ID of current commit for current brach
+
+# This retrieves hashes for all the commits which have a tag (just releases (?))
+# taglists <- gert::git_tag_list()
+
+# How to get a tash for a specific version
+# hash136 <- taglists[str_detect(taglists$name, "1.3.6"), "commit"]
+
+# Keep a list with just the benchmark outputs
+benchmarks_outputs <- stored_benchmarks[["benchmarks"]]
+
+# In order to bind and identify all dataframes within benchmark_outputs
+# the list needs to be named, so I'm naming it with the hashs
+# Then this argument will be used bind rows
+commit_hash <- stored_benchmarks[["commit_hash"]]
+
+# Name list elements
+names(benchmarks_outputs) <- commit_hash
+
+# bind rows and add .id
+benchmarks_df <- dplyr::bind_rows(benchmarks_outputs, .id = "hash")
+```
+
+<!-- This css stuff is just to enable rmarkdown to use a wider area of screen -->
+
+<style type="text/css">
+.main-container {
+  max-width: 1800px;
+  margin-left: auto;
+  margin-right: auto;
+}
+</style>
+
+## Execution times
+
+
+
+```{r echo=FALSE, message=FALSE, warning=FALSE}
+
+ggplotly(
+  benchmarks_df  %>% 
+           ggplot(aes(x = time, y = mean))+
```

Comment:
Instead of having time as the x-axis, what do you think of changing it to the commit ?

Reply:
I've added a plot with the hashes in the x-axis, just to showcase the result of both kinds of plot, I prefer the one with the dates, but is up to you.